Rocket.Chat/Extensions/API/APIExtensions.swift

```diff
@@ -9,6 +9,8 @@
 import Foundation
 
 extension API {
+    static var current: API?
```

Comment:
@cardoso @rafaelks I added this property for us to reuse the current instance of `API`, it avoids having unnecessary instances in memory. Since we always call for `current` function below and it updates all the relevant information related to an `API` instance it seems that it will not cause any issue. 

Do you guys agree with this? Or am I missing some scenario where we can't share the same instance of an `API`? 


I'm asking it because I've found another way to solve the API related memory leaks, but when I remove this "single instance control" we wind up having always two different API instances in memory. They are not leaked because they're being referenced by `SocketManager.sharedInstace` which is always live throughout the application's lifecycle, but I think that we only need one instance.

Reply:
@filipealva I believe it's fine for us to have a `current` property being shared all over the app, but I don't see any place you clearing this variable. Should we do that on changing servers at least?

Reply:
@rafaelks That's a good point. Actually, I think that this `current` can be a memory cache of our `API` instance throughout the application's lifecycle. It doesn't need to be cleaned because always when `current()` is called we refresh all of its properties, it should be a private (I would make it private on the next commit) property of `API` for memory caching.

I didn't think about it with the architecture point of view when I come up with this idea, I was just trying to isolate/solve the memory leaks. 

Maybe I just send the fix for the root memory leak and we take note of this multiple API instances and create a separate issue only for thinking about the best architectural approach, sounds good?